taskWithList/task/task/list.c

```diff
@@ -0,0 +1,72 @@
+#include "list.h"
+#include <stdlib.h>
+#include <stdbool.h>
+#include <assert.h>
+#include <stdio.h>
+
+typedef struct ListElement {
+	Value value;
+	ListElement* next;
+} ListElement;
+
+typedef struct List {
+	ListElement* head;
+} List;
+
+List* createList() {
```

Comment:
```suggestion
List* createList(void) {
```
По идее компилятор должен был предупреждение выдавать, что сигнатура функции не совпадает с сигнатурой из заголовочного файла.